homeassistant/components/home_connect/coordinator.py

```diff
@@ -0,0 +1,227 @@
+            self.hass,
+            self._event_listener(),
+            f"home_connect-events_listener_task-{self.config_entry.entry_id}",
+        )
+
+    async def _event_listener(self) -> None:
+        """Match event with listener for event type."""
+        while True:
+            try:
+                async for event_message in self.client.stream_all_events():
+                    match event_message.type:
+                        case EventType.STATUS:
+                            statuses = self.data[event_message.ha_id].status
+                            for event in event_message.data.items:
+                                if (
+                                    event.key not in StatusKey
+                                    or event.key is EventKey.UNKNOWN
+                                ):
+                                    continue
+                                status_key = StatusKey(event.key)
+                                if status_key in statuses:
+                                    statuses[status_key].value = event.value
+                                else:
+                                    statuses[status_key] = Status(
+                                        status_key, event.value
+                                    )
+                            await self._call_event_listener(event_message)
+
+                        case EventType.NOTIFY:
+                            settings = self.data[event_message.ha_id].settings
+                            for event in event_message.data.items:
+                                if (
+                                    event.key not in SettingKey
+                                    or event.key is EventKey.UNKNOWN
+                                ):
+                                    continue
+                                setting_key = SettingKey(event.key)
+                                if setting_key in settings:
+                                    settings[setting_key].value = event.value
+                                else:
+                                    settings[setting_key] = GetSetting(
+                                        setting_key, event.value
+                                    )
+                            await self._call_event_listener(event_message)
+
+                        case EventType.EVENT:
+                            await self._call_event_listener(event_message)
+
+            except (EventStreamInterruptedError, HomeConnectRequestError) as error:
+                _LOGGER.debug(
+                    "Non-breaking error (%s) while listening for events,"
+                    " continuing in 30 seconds",
+                    type(error).__name__,
+                )
+                await asyncio.sleep(EVENT_STREAM_RECONNECT_DELAY)
+            except HomeConnectApiError as error:
+                _LOGGER.error("Error while listening for events: %s", error)
+                break
```

Comment:
Here we probably want to reload the integration.

Reply:
How can we reload the integration from there?

Reply:
If we have access to the config entry we can call `hass.config_entries.async_reload`. Example:

https://github.com/home-assistant/core/blob/b8632063f5d86931393abdc9840a41831a211470/homeassistant/components/matter/__init__.py#L187-L189

Reply:
Should I create the task also? or can I call `hass.config_entries.async_reload` directly?

Reply:
Use this helper:

https://github.com/home-assistant/core/blob/75bdcee3e4921410485e3a99fbc8596036dfc5c6/homeassistant/config_entries.py#L2069